components/schema-migrator/migrations/director/20231020141132_add-entity-type-and-make-part-of-products-optional-for-packages.up.sql

```diff
@@ -0,0 +1,205 @@
+       elements.value      AS value
+FROM entity_types,
+     jsonb_each(entity_types.documentation_labels) AS expand,
+     jsonb_array_elements_text(expand.value) AS elements;
+
+
+DROP VIEW IF EXISTS entity_types_tenants;
+
+CREATE VIEW entity_types_tenants AS
+SELECT e.*, ta.tenant_id, ta.owner FROM entity_types AS e
+                                            INNER JOIN tenant_applications ta ON ta.id = e.app_id;
+
+DROP VIEW IF EXISTS tenants_entity_types;
+
+CREATE OR REPLACE VIEW tenants_entity_types
+            (tenant_id, formation_id, id, ord_id, app_id, local_id, level, title, short_description, description, system_instance_aware, 
+            changelog_entries, package_id, visibility, links, part_of_products, policy_level,
+            custom_policy_level, release_status, sunset_date, successors, extensible_supported, extensible_description, tags, labels, 
+            documentation_labels, resource_hash, version_value, version_deprecated, version_deprecated_since, version_for_removal)
+AS
+SELECT DISTINCT t_apps.tenant_id,
+                t_apps.formation_id,
+                et.id,
+                et.ord_id,
+                et.app_id,
+                et.local_id,
+                et.level,
+                et.title,
+                et.short_description,
+                et.description,
+                et.system_instance_aware,
+                et.changelog_entries,
+                et.package_id,
+                et.visibility,
+                et.links, 
+                et.part_of_products,
+                et.policy_level,
+                et.custom_policy_level,
+                et.release_status,
+                et.sunset_date,
+                et.successors,
+                actions.supported,
+                actions.description,
+                et.tags,
+                et.labels,
+                et.documentation_labels,
+                et.resource_hash,
+                et.version_value,
+                et.version_deprecated,
+                et.version_deprecated_since,
+                et.version_for_removal
+FROM entity_types et
+         JOIN (SELECT a1.id,
+                      a1.tenant_id AS tenant_id,
+                      'aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa' AS formation_id
+               FROM tenant_applications a1
+               UNION ALL
+               SELECT apps_subaccounts.id,
+                      apps_subaccounts.tenant_id,
+                      apps_subaccounts.formation_id
+               FROM apps_subaccounts
+               UNION ALL
+               SELECT apps_subaccounts.id,
+                      apps_subaccounts.tenant_id,
+                      'aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa' AS formation_id
+               FROM apps_subaccounts) t_apps
+              ON et.app_id = t_apps.id,
+              jsonb_to_record(et.extensible) actions(supported text, description text);
+
+CREATE INDEX entity_types_app_template_version_id ON entity_types (app_template_version_id);
```

Comment:
Can you add one more index on the `app_id` because it will affect positively the ORD Service performance.

Heads up to drop it in the down migration.